src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/index.jsx

```diff
@@ -505,24 +516,47 @@ class GlobalSelectionHeader extends React.Component {
       ? [parseInt(forceProject.id, 10)]
       : this.props.selection.projects;
 
-    const [projects, nonMemberProjects] = this.getProjects();
-
+    const [memberProjects, nonMemberProjects] = this.getProjects();
     return (
       <Header className={className}>
         <HeaderItemPosition>
           {shouldForceProject && this.getBackButton()}
-          <MultipleProjectSelector
-            organization={organization}
-            shouldForceProject={shouldForceProject}
-            forceProject={forceProject}
-            projects={projects}
-            loadingProjects={loadingProjects}
-            nonMemberProjects={nonMemberProjects}
-            value={this.state.projects || this.props.selection.projects}
-            onChange={this.handleChangeProjects}
-            onUpdate={this.handleUpdateProjects}
-            multi={this.hasMultipleProjectSelection()}
-          />
+          <Projects orgId={organization.slug} limit={PROJECTS_PER_PAGE} globalSelection>
+            {({projects, initiallyLoaded, hasMore, onSearch, fetching}) => {
+              const paginatedProjectSelectorCallbacks = {
+                onScroll: ({clientHeight, scrollHeight, scrollTop}) => {
+                  if (
+                    !fetching &&
+                    scrollTop + clientHeight >= scrollHeight - clientHeight
+                  ) {
+                    onSearch(this.state.searchQuery, {append: true});
```

Comment:
Should this use the debounced wrapper incase people scroll up and down fast?

Reply:
hmmm upon further testing I think I will revert to not using the debounced wrapper. With the wrapper, the user has to wait 200 ms before the selector loads additional projects, but I feel like aggressively loading the projects before the user reaches the bottom feels a bit better. The scrolling up and down should hopefully not be a large issue as I check to make sure the `<Projects>` component is not currently fetching anything before starting a new fetch. 

If the debounce is still preferred perhaps I could switch to a leading debounce which lets in the first call and debounces the following. Let me know what is preferred here.

Reply:
A leading debounce could work, I was concerned about users quickly scrolling up and down and queuing multiple requests up.

Reply:
gotcha, i dont think it should be a huge problem since I make sure to not fetch additional projects if there is a request going on, but it isn't a guarantee if the props don't refresh fast enough so I'll add in the leading debounce.